File: class2relational/src/main/java/org/atlanmod/model/dynamic/classModel/ClassToAttributes.java
package org.atlanmod.model.dynamic.classModel;

import org.atlanmod.model.dynamic.DynamicElement;

import java.util.ArrayList;
import java.util.List;

public class ClassToAttributes extends ClassLink {

    public ClassToAttributes(ClassClass source, List<ClassAttribute> target) {
        super(ClassModel.CLASS_ATTRIBUTES, source, null);
        this.setTarget(new ArrayList<>(target));
    }

    public String toString() {
        StringBuilder target_ids = new StringBuilder("[");
        for (int i=0; i<getTarget().size(); i++)
            target_ids.append(((ClassElement) this.getTarget().get(i)).getId()).append(i == getTarget().size() - 1 ? "]" : ",");
        return "(" + ((ClassElement) this.getSource()).getId() + ", " + super.getType() + ", " + target_ids + ")";
    }

}

File: sparktl-transformations/src/main/scala/org/atlanmod/class2relational/model/classmodel/ClassToAttributes.scala
package org.atlanmod.class2relational.model.classmodel

class ClassToAttributes (source: ClassClass, target: List[ClassAttribute])
  extends ClassLink(ClassMetamodel.CLASS_ATTRIBUTES, source, target){

    def this(source: ClassClass, target: ClassAttribute) =
        this(source, List(target))

    override def getSource: ClassClass = source
    override def getTarget: List[ClassAttribute] = target

    override def toString: String =
        "(" + source.getId + ", " + getType + ", " + target.map(a => a.getId).mkString("[", ", ", "]") + ")"

}
